### PR TITLE
FISTA iterative reconstruction

### PR DESCRIPTION
**Pull request by dkazanc**
- A wrapper for FISTA reconstruction method in [TomoPhantom](https://github.com/dkazanc/TomoPhantom) added  
- The algorithm also depends on the [CCPi Regularisation Toolkit](https://github.com/vais-ral/CCPi-Regularisation-Toolkit)
- Added jenkins tests.
p.s. I've installed both dependencies in `base` conda environment.

### CODE REVIEW

**Comment by dkazanc** [comment]
Actually let's wait with merging this.  I've decided to move FISTA from TomoPhantom to [FISTA-tomo](https://github.com/dkazanc/FISTA-tomo) and wrap it with conda-build. I think it makes sense to keep it separate.

**Comment by dkazanc** [comment]
Working fine now from FISTA-tomo and can be merged. The dependency can be conveniently installed with `conda install -c dkazanc fista-tomo` . The `base` conda savu updated.

File: savu/plugins/reconstructions/fista_recon.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
+# Copyright 2014 Diamond Light Source Ltd.
```
Comment (markbasham) [inline savu/plugins/reconstructions/fista_recon.py]: Probably worth updating this to 2018

File: savu/plugins/reconstructions/fista_recon.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+.. moduleauthor:: Daniil Kazantsev <scientificsoftware@diamond.ac.uk>
+"""
+
+from savu.plugins.reconstructions.base_recon import BaseRecon
+from savu.data.plugin_list import CitationInformation
+from savu.plugins.driver.gpu_plugin import GpuPlugin
+
+import numpy as np
+# install FISTA-tomo with: conda install -c dkazanc fista-tomo
```
Comment (markbasham) [inline savu/plugins/reconstructions/fista_recon.py]: Probably a todo for the future, but we should move this to a DLS or CCPi repo just for longevity

File: savu/plugins/reconstructions/fista_recon.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+        self.regularisation = self.parameters['regularisation']
+        self.regularisation_parameter = self.parameters['regularisation_parameter']
+        self.regularisation_parameter2 = self.parameters['regularisation_parameter2']
+        self.regularisation_iterations = self.parameters['regularisation_iterations']
+        self.time_marching_parameter = self.parameters['time_marching_parameter']
+        self.edge_param = self.parameters['edge_param']
+        self.NDF_penalty = self.parameters['NDF_penalty']
+        self.Rectools = None
+
+        # extract data and data-related parameters
```
Comment (markbasham) [inline savu/plugins/reconstructions/fista_recon.py]: Lots of comments here, probably worth a tidy up

File: savu/plugins/reconstructions/fista_recon.py [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+                                    regularisation_parameter = self.regularisation_parameter,\
+                                    regularisation_iterations = self.regularisation_iterations,\
+                                    regularisation_parameter2 = self.regularisation_parameter2,\
+                                    time_marching_parameter = self.time_marching_parameter,\
+                                    edge_param = self.edge_param,\
+                                    NDF_penalty = self.NDF_penalty,\
+                                    lipschitz_const = self.Lipschitz_const)
+        return recon
+
+    def get_value(self):
```
Comment (markbasham) [inline savu/plugins/reconstructions/fista_recon.py]: Given this use, i would probably call this something like setup_Lipschitz_constant

File: savu/test/jenkins/plugin_tests/filter_tests/fista_recon_test.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+# Copyright 2014 Diamond Light Source Ltd.
```
Comment (markbasham) [inline savu/test/jenkins/plugin_tests/filter_tests/fista_recon_test.py]: Worth a quick update to 2018

**Comment by dkazanc** [comment]
Thanks Mark, will fix the remarks and merge. I've also just implemented Ordered-Subsets for FISTA so going to add this as well.

**Pull request merged**